rest/route/distro.go

```diff
@@ -150,10 +157,14 @@ func (h *distroIDPutHandler) Parse(ctx context.Context, r *http.Request) error {
 // (b) creates a new resource based on the Request-URI and JSON payload
 func (h *distroIDPutHandler) Run(ctx context.Context) gimlet.Responder {
 	user := MustHaveUser(ctx)
-	original, err := data.FindDistroById(h.distroID)
-	if err != nil && err.(gimlet.ErrorResponse).StatusCode != http.StatusNotFound {
-		return gimlet.MakeJSONErrorResponder(errors.Wrapf(err, "Database error for find() by distro id '%s'", h.distroID))
+	original, err := distro.FindOne(distro.ById(h.distroID))
+	if err != nil && !adb.ResultsNotFound(err) {
```

Comment:
We don't need to check ResultsNotFound anymore since we can just check if original == nil (should update distroExists for that as well)